bindings/rust/bench/historical-perf/bench-past.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# immediately bail if any command fails
+set -e
+
+# suppress stdout and most cargo warnings
+exec >/dev/null
+export CARGO_TERM_QUIET=true
+export RUSTFLAGS=-Awarnings
+
+# go to s2n-tls/bindings/rust/bench/
+pushd "$(dirname "$0")"/../
+bench_path="$(pwd)"
+
+# delete past runs
+rm -rf target/historical-perf
+
+# make Cargo.toml point s2n-tls to the cloned old version
+sed -i "s|s2n-tls = .*|s2n-tls = { path = \"target/s2n-tls/bindings/rust/s2n-tls\" }|" Cargo.toml 
+
+# ensure Cargo.toml gets changed back on exit; retains original exit status
+trap "{ status=$?; sed -i 's|s2n-tls = .*|s2n-tls = { path = \"../s2n-tls\" }|' $bench_path/Cargo.toml; exit $status; }" EXIT
```

Comment:
Ahh I missed this. Modifying the source is pretty hacky. If you really want to do this then I would suggest a `Cargo.template` which you modify and we can only run the project from the shell scripts.

Reply:
How would the Cargo.template work? Is it just a copy of the Cargo.toml that's modified with the new path? Or is it a special cargo feature?

Reply:
You can specify a different `Cargo.toml`, but `--manifest-path` requires the different toml file to be called exactly `Cargo.toml`, so it'd need to be stored in a different directory. However, storing the `Cargo.toml` in a different directory changes all of the path in the `Cargo.toml` since it's all relative to where the `Cargo.toml` lives. I really couldn't find a better option. Comment above talking about this: https://github.com/aws/s2n-tls/pull/4083#discussion_r1258676206

Also, I wanted everything to run with `cargo bench` without a separate run script, since it makes the most sense for the user experience. Yes this is a little hacky, but I think it's well worth the ease of use, and it seems to be the only way to do it.